### ISSUE TITLE
Silent error when an email template is not found

### ISSUE THREAD

**Issue by shavounet** [magento/magento2#8437]
### Preconditions
1. Tested on Magento CE 2.1.3 with PHP 7

### Steps to reproduce
1. Create a custom controller, sending a custom mail, [mostly like this](http://magento.stackexchange.com/questions/150728/sending-custom-emails-in-magento-2)
2. Make a mistake in the template file name
3. Try to send the mail

### Expected result
1. The mail is correctly sent with content

### Actual result
1. The mail is correctly sent but the body is empty

IMO, `\Magento\Email\Model\AbstractTemplate::loadDefault` should throw an error (or **at least** log an error message...), when the file is not found. In this case `$rootDirectory->getRelativePath($templateFile)` returns an empty string and so does `$rootDirectory->readFile('')`... Maybe the problem is deeper, but a silent error should not happen.

If it matters, my send code in the controller was followed by a redirect (after filling a form).

## Best way to reproduce
1. Go to file `app/code/Magento/Customer/etc/email_templates.xml`
2. Make a mistake in the template file name: replace from `account_new.html` to `account_new1.html` in tag:
```
...
    <template id="customer_create_account_email_template" label="New Account" file="account_new1.html" type="html" module="Magento_Customer" area="frontend"/>
...
```
3. Create new customer on frontend

### Expected result
1. The email about new customer not sent.
2. Exception like "Can not find the file account_new1.htm" is present in logs

### Actual result
1. The mail is correctly sent but the body is empty
2. Any exception is absent in logs.

**Comment by vrann** [comment]
@shavounet I see the following code in the logic of the filesystem adapter which should throw an exception in case if you misspelled the name of the file:
https://github.com/magento/magento2/blob/develop/lib/internal/Magento/Framework/Filesystem/Driver/File.php#L149
Do you see why doesn't it throw it?
Can you provide the code snippet of your controller sending email, please?

**Comment by shavounet** [comment]
I don't have much time to test it again, but here is the main method.

```php
    /**
     * Send the quote email
     * @param Quote $quote
     * @throws \Exception
     */
    public function sendNewQuoteEmail($quote)
    {
        $this->inlineTranslation->suspend();

        $receiverEmail = $this->scopeConfig->getValue('vendor_personalization/quote/contact_email', ScopeInterface::SCOPE_STORES);

        if(empty($receiverEmail)) {
            throw new \Exception('The receiver email is undefined in Vendor configuration');
        }

        $this->transportBuilder
            // Set required data
            ->setTemplateIdentifier('vendor_pms_new_quote')
            ->setTemplateOptions([
                'area' => \Magento\Framework\App\Area::AREA_FRONTEND,
                'store' => $this->storeManager->getStore()->getId()
            ])
            ->setTemplateVars(['quote' => $quote])
            ->setFrom([
                'name' => $quote->getFullName(),
                'email' => $quote->getEmail(),
            ])
            ->addTo($receiverEmail)

            // Send message
            ->getTransport()
            ->sendMessage();

        $this->inlineTranslation->resume();
    }
```

**Comment by vrann** [comment]
@shavounet thank you let me try

**Comment by magento-engcom-team** [comment]
@shavounet , thank you for your report.
Unfortunately, link from provided steps to reproduce returns 404.
We were not able to reproduce this issue. Please provide more detailed steps to reproduce or try to reproduce this issue on a clean installation or latest release.

**Comment by shavounet** [comment]
Hello @magento-engcom-team,
The implementation details are provided in my 2nd comment, it's a very simple implementation. I don't have a test environment anymore to provide more feedback, but did you try it out ? I have to admit I agree with @vrann ; it shouldn't fail... yet it did

Maybe there is some side effect with other methods ? (getScheme, getAbsolutePath, ...) If you look at those examples, some case (2nd and 3rd) can be problematic :
```
php > $a = @file_get_contents('');
php > var_dump($a);
bool(false)
php > $a = @file_get_contents('/');
php > var_dump($a);
string(0) ""
php > $a = @file_get_contents('file://');
php > var_dump($a);
string(0) ""
```

**Comment by magento-engcom-team** [comment]
Update description:
### Preconditions
1. Tested on Magento CE 2.1.3 with PHP 7

### Steps to reproduce
1. Create a custom controller, sending a custom mail, [mostly like this](http://magento.stackexchange.com/questions/150728/sending-custom-emails-in-magento-2)
2. Make a mistake in the template file name
3. Try to send the mail

### Expected result
1. The mail is correctly sent with content

### Actual result
1. The mail is correctly sent but the body is empty

IMO, `\Magento\Email\Model\AbstractTemplate::loadDefault` should throw an error (or **at least** log an error message...), when the file is not found. In this case `$rootDirectory->getRelativePath($templateFile)` returns an empty string and so does `$rootDirectory->readFile('')`... Maybe the problem is deeper, but a silent error should not happen.

If it matters, my send code in the controller was followed by a redirect (after filling a form).

## Best way to reproduce
1. Go to file `app/code/Magento/Customer/etc/email_templates.xml`
2. Make a mistake in the template file name: replace from `account_new.html` to `account_new1.html` in tag:
```
...
    <template id="customer_create_account_email_template" label="New Account" file="account_new1.html" type="html" module="Magento_Customer" area="frontend"/>
...
```
3. Create new customer on frontend

### Expected result
1. The email about new customer not sent.
2. Exception like "Can not find the file account_new1.htm" is present in logs

### Actual result
1. The mail is correctly sent but the body is empty
2. Any exception is absent in logs.

**Comment by magento-engcom-team** [comment]
@shavounet, thank you for your report.
We've created internal ticket(s) MAGETWO-83674 to track progress on the issue.

**Comment by okorshenko** [comment]
Hi @shavounet 
The issue has been fixed and delivered in 2.2-develop branch. Will be available with upcoming patch release.
